app/src/main/java/com/infomaniak/swisstransfer/ui/components/TwoPaneScaffold.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.infomaniak.swisstransfer.ui.components
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.material3.adaptive.ExperimentalMaterial3AdaptiveApi
+import androidx.compose.material3.adaptive.WindowAdaptiveInfo
+import androidx.compose.material3.adaptive.layout.AnimatedPane
+import androidx.compose.material3.adaptive.layout.ListDetailPaneScaffold
+import androidx.compose.material3.adaptive.layout.calculatePaneScaffoldDirective
+import androidx.compose.material3.adaptive.navigation.ThreePaneScaffoldNavigator
+import androidx.compose.material3.adaptive.navigation.rememberListDetailPaneScaffoldNavigator
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.window.core.layout.WindowHeightSizeClass
+
+/**
+ * A composable function that sets up a List-Detail interface using a three-pane scaffold navigator.
+ * This function adapts its layout based on the given window size class and manages the navigation
+ * between list and detail panes.
+ *
+ * @param T The type parameter representing the data model used in the navigator.
+ * @param windowAdaptiveInfo An instance of [WindowAdaptiveInfo] that provides information about
+ * the current window's size class and adaptive state.
+ * @param listPane A composable function that describes the content of the list pane.
+ * It is provided with an instance of [ThreePaneScaffoldNavigator].
+ * @param detailPane A composable function that describes the content of the detail pane.
+ * It is provided with an instance of [ThreePaneScaffoldNavigator].
+ * @param modifier A [Modifier] for this composable. Defaults to [Modifier].
```

Comment:
Is all of these details really needed? Because most of these parameter descriptions are self explanatory, the comment is pretty much redundant and adds a lot of unnecessary noise in the file. 

The only parameter that could make sens to explain would be the generic type T. But even then, this parameter T could probably use a little bit more explanations as it only states what's written in the signature. Currently, it doesn't help understand what T's purpose is which should probably be the comment's main point.

Reply:
By the way, what's the purpose of this T value?